### DISCUSSION TITLE
Ship decorators

### DISCUSSION THREAD

**Discussion by gegoune** [nvim-tree/nvim-tree.lua#3036] [General]
Hi @alex-courtis,

what do you think about providing mechanism to enable built-in decorators. We could then ship and maintain some, like quickfix one from https://github.com/nvim-tree/nvim-tree.lua/wiki/Decorators for example. Perhaps more will come in the future.
This would allow users to enable them either through api call or configuration.

All the best to you and your close ones during holiday season!

**Comment by alex-courtis** [comment]
That would work nicely; a library of optional, maybe contributed, decorators.

Right now, everything is enabled via `renderer.decorators` but new ones don't have to be added.

API would be useful to, say, toggle one on or off. I couldn't come up with anything elegant at the time.

HNY!
